src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -221,8 +234,12 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
         Source
           .fromIterator(() => pages)
           .mapAsync[tezosNodeOperator.BlockFetchingResults](1)(identity)
-          .mapAsync(1) {
-            processBlocksPage(_) flatTap (_ => processTezosAccountsCheckpoint >> processTezosDelegatesCheckpoint)
+          .mapAsync(1) { fetchingResults =>
+             processBlocksPage(fetchingResults).flatMap { result =>
```

Comment:
If we don't need the "intermediate results" of the flatMap, _flatTap_ is the good choice.

Here what needs be reported is the `processBlocksPage` result, hence we can flatTap over backing and endorsing rights, as well as checkpoints, removing the need to keep track of the intermediate `result` to map it back at the end (flatTap is exactly doing that for us).

Reply:
What if the backing/endorse rights fails to correctly process?

We actually have a similar issue with voting data (which is currently being partially removed, luckily).

The point is mainly about transactional guarantees of consistency for the whole processing. Once blocks (and tightly-related data) has been atomically stored, any failure in subsequent processing of corollary data might fail, and Lorre will not pick it up at next retried run.
That's the whole reason for checkpointing accounts/delegates ids, so that any failure will not lose that information anyway, which will be picked up by those two instructions right here at line 240.

I'm only pointing this out here by chance, this is a more general issue of consistency that will continue to threaten the lorre process as we add more and more data, food for future thought.
I'm hereby saying that we need to (re-)think the way data is stored, to manage a safe way to not risk losing any data due to complex elaboration. One approach is to do multi-stage processing, saving data as soon and as raw as possible, and then parse and process it in an immediate later stage. Then when can clean the raw data only if it's been confirmed as processed.

Reply:
ok, used flatTap, also, there is a lot of places it could fail and we could not retrieve data elsewhere. For example I was testing my changes in the different branch few days ago and I stopped Lorre in some unfortunate moment that did not allow to continue fetching so the easiest solution was to remove DB and start again.

Reply:
In general this shouldn't happen. The system should be able to stop in-between blocks' processing and never lose data, apart from:
- voting information
- rights to endorsements/baking (as of this PR)

Reply:
If I encounter it again I'll msg you. If I remember correctly the problem was with violating primary key in some table - Lorre crashed.